File: java-demo/src/main/java/demo1/Ex2.java
package demo1;

import java.util.stream.IntStream;

import akka.actor.ActorSystem;
import akka.stream.ActorMaterializer;
import akka.stream.javadsl.Flow;
import akka.stream.javadsl.RunnableGraph;
import akka.stream.javadsl.Sink;
import akka.stream.javadsl.Source;
import scala.concurrent.Future;
import scala.runtime.BoxedUnit;

public class Ex2 {
  public static void main(String[] args) {
    ActorSystem system = ActorSystem.create("demo1");
    ActorMaterializer materializer = ActorMaterializer.create(system);

    // Create a source with some integer and print them
    IntStream stream = IntStream.range(1, 4);
    Source<Integer, ?> source = Source.from(() -> stream.iterator());
    Flow<Integer, Integer, ?> f = Flow.<Integer> create().map(x -> x + 10).filter(x -> x % 2 == 0);
    Sink<Integer, Future<BoxedUnit>> sink = Sink.foreach(x -> System.out.println(x));
    RunnableGraph<?> runnable = source.via(f).to(sink);
    runnable.run(materializer);
  }
}

File: scala-demo/src/main/scala/demo1/Ex2.scala
package demo1

import akka.actor.ActorSystem
import akka.stream.ActorFlowMaterializer
import akka.stream._
import akka.stream.scaladsl._

import scala.concurrent.Future
import scala.concurrent.Await
import scala.concurrent.duration.Duration
import scala.concurrent.duration._

object Ex2 extends App {
  implicit val system = ActorSystem("demo1")
  implicit val materializer = ActorFlowMaterializer()
  import system.dispatcher

  // Create a stream with a mapping stage (using a Flow)
  val source = Source(1 to 3)
  val flow = Flow[Int].map(_ + 10).filter(_ % 2 == 0)
  val sink = Sink.foreach(println)
  source.via(flow).runWith(sink)
}